src/CMakeLists.txt

```diff
@@ -114,6 +114,7 @@ add_subdirectory(ringct)
 add_subdirectory(checkpoints)
 add_subdirectory(cryptonote_basic)
 add_subdirectory(cryptonote_core)
+add_subdirectory(lmdb)
```

Comment:
That said, some of the stuff in blockchain_blackball does rely on lmdb too IIRC.

Reply:
The light wallet server is not abstracted to work with multiple database types. I didn't consider this an issue; I thought Monero already required LMDB as a dependency. Does the core code still work with BerkeleyDB for instance?

Reply:
The bdb backend is obsolete and unmaintained, that could be removed. Whether LMDB becomes sole dep might be a question to ask though. At the moment, all still goes through the BlockchainDB class (except some tools which read direct off the (LMDB) raw db).

Reply:
There doesn't appear to be a way to build without LMDB currently, the vendored version in `external/` is always compiled in.

I thought about the abstractions some, and it should be possible to get the light wallet server working without LMDB. This would likely be a static interface - CMake would include different headers and compile against different files. The server was heavily optimized for LMDB though, so it would be interesting.

Reply:
I'm OK with it needing lmdb if it's a conscious choice.